File: backups/2025-01-29T14-26-12-685Z/backend/migrate.py
import psycopg2
import os
from dotenv import load_dotenv
from pathlib import Path

load_dotenv()

def migrate():
    conn = psycopg2.connect(
        host=os.getenv("VITE_DB_HOST"),
        database=os.getenv("VITE_DB_NAME"),
        user=os.getenv("VITE_DB_USER"),
        password=os.getenv("VITE_DB_PASSWORD"),
        port=os.getenv("VITE_DB_PORT"),
        sslmode='require'
    )
    
    try:
        cur = conn.cursor()
        
        # Create migrations table if it doesn't exist
        cur.execute("""
            CREATE TABLE IF NOT EXISTS migrations (
                id SERIAL PRIMARY KEY,
                name VARCHAR(255) NOT NULL,
                executed_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
            );
        """)
        conn.commit()
        
        # Get list of migration files
        migrations_dir = Path(__file__).parent / 'migrations'
        migration_files = sorted(migrations_dir.glob('*.sql'))
        
        for migration_file in migration_files:
            # Check if migration has been executed
            cur.execute(
                "SELECT id FROM migrations WHERE name = %s",
                (migration_file.name,)
            )
            if not cur.fetchone():
                print(f"Executing migration: {migration_file.name}")
                
                # Read and execute migration
                with open(migration_file) as f:
                    sql = f.read()
                    cur.execute(sql)
                
                # Record migration
                cur.execute(
                    "INSERT INTO migrations (name) VALUES (%s)",
                    (migration_file.name,)
                )
                conn.commit()
                print(f"Completed migration: {migration_file.name}")
            else:
                print(f"Skipping migration {migration_file.name} - already executed")
        
        print("All migrations completed successfully")
        
    except Exception as e:
        print("Migration failed:", str(e))
        conn.rollback()
        raise
    finally:
        conn.close()

if __name__ == "__main__":
    migrate()

File: backups/2025-01-29T14-26-12-685Z/scripts/migrate.js
import pkg from 'pg';
const { Client } = pkg;
import { readFileSync } from 'fs';
import { fileURLToPath } from 'url';
import { dirname, join } from 'path';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

const config = {
  host: '104.154.177.118',
  database: 'postgres',
  user: 'postgres',
  password: '19910331',
  port: 5432,
  ssl: {
    rejectUnauthorized: false // Required for GCP Cloud SQL
  }
};

async function migrate() {
  const client = new Client(config);

  try {
    await client.connect();
    console.log('Connected to database');

    // Create migrations table if it doesn't exist
    await client.query(`
      CREATE TABLE IF NOT EXISTS migrations (
        id SERIAL PRIMARY KEY,
        name VARCHAR(255) NOT NULL,
        executed_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
      );
    `);

    // Read and execute migration files
    const migrations = [
      'create_auth_schema.sql',
      'create_assessments.sql',
      'create_forum.sql'
    ];

    for (const migration of migrations) {
      const exists = await client.query(
        'SELECT id FROM migrations WHERE name = $1',
        [migration]
      );

      if (exists.rows.length === 0) {
        const sql = readFileSync(join(__dirname, '..', 'migrations', migration), 'utf8');
        console.log(`Executing migration: ${migration}`);
        await client.query(sql);
        await client.query(
          'INSERT INTO migrations (name) VALUES ($1)',
          [migration]
        );
        console.log(`Completed migration: ${migration}`);
      } else {
        console.log(`Skipping migration ${migration} - already executed`);
      }
    }

    console.log('All migrations completed successfully');
  } catch (error) {
    console.error('Migration failed:', error);
    process.exit(1);
  } finally {
    await client.end();
  }
}

migrate().catch(console.error);